engine/src/flutter/lib/web_ui/lib/ui_web/src/ui_web/platform_view_registry.dart

```diff
@@ -12,7 +13,19 @@ typedef ParameterizedPlatformViewFactory = Object Function(int viewId, {Object?
 typedef PlatformViewFactory = Object Function(int viewId);
 
 /// The platform view registry for this app.
-final PlatformViewRegistry platformViewRegistry = PlatformViewRegistry();
+PlatformViewRegistry get platformViewRegistry => _platformViewRegistry ??= PlatformViewRegistry();
+PlatformViewRegistry? _platformViewRegistry;
+
+/// Overrides the platform view registry.
+///
+/// If [value] is null, resets the value of [platformViewRegistry] back to real
+/// implementation.
+///
+/// This is intended for tests only.
+@visibleForTesting
+void debugOverridePlatformViewRegistry(PlatformViewRegistry? value) {
+  _platformViewRegistry = value;
+}
```

Comment:
Instead of throwing away the real platform registry, let's keep it around and restore it:
```suggestion
late final PlatformViewRegistry? _defaultPlatformViewRegistry = PlatformViewRegistry();
PlatformViewRegistry? _platformViewRegistryOverride;

PlatformViewRegistry get platformViewRegistry =>
    _platformViewRegistryOverride ?? _defaultPlatformViewRegistry;

/// Overrides the platform view registry.
///
/// If [value] is null, resets the value of [platformViewRegistry] back to real
/// implementation.
///
/// This is intended for tests only.
@visibleForTesting
void debugOverridePlatformViewRegistry(PlatformViewRegistry? value) {
  _platformViewRegistryOverride = value;
}
```